gdal/src/main/scala/geotrellis/contrib/vlm/gdal/GDALDataPath.scala

```diff
@@ -43,9 +42,26 @@ case class GDALDataPath(
   path: String,
   compressedFileDelimiter: String = "!"
 ) extends DataPath {
-
   import Schemes._
-  import Patterns._
+
+  val isVSIFormatted: Boolean =
+    path.startsWith("/vsi")
+
+  // scala-uri has problems encoding/decoding certain characters,
+  // so to account for that, we check to see if the path can
+  // be parsed at all. If it can't then we use the UrlPath
+  // to encode the path so that it can be used by the
+  // classes in the library
+  private val gdalPath =
+    if (isVSIFormatted)
+      VSIPath(path)
+    else
+      Url.parseOption(path) match {
+        case Some(uri) => URIPath(uri)
+        case None =>
+          val encodedPath = UrlPath.fromRaw(path).toString
+          URIPath(Url.parse(encodedPath))
+      }
 
   private val badPrefixes: List[String] =
     List("gt+", "gtiff+")
```

Comment:
This is a pretty general comment but I think its worth mentioning here. The idea that while parsing `GDAL` path we have a list of things that we know are bad goes against the general design principle of [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns). 

What happens when more prefixes are added, does the logic in this class become invalid? And why would adding a new reading mechanism, that is unrelated to GDAL would require changing class concerned only with GDAL paths. These prefixes do serve a function to disambiguate URIs that can be parsed by multiple readers but such checking should happen outside of reader specific code and where the ambiguity exists, delegating the full parsing to the appropriate module/class.